core/templates/pages/blog-dashboard-page/blog-dashboard-page.component.spec.ts

```diff
@@ -64,17 +86,53 @@ describe('Blog Dashboard Page Component', () => {
     fixture = TestBed.createComponent(BlogDashboardPageComponent);
     component = fixture.componentInstance;
     urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    mockWindowRef = TestBed.inject(WindowRef) as unknown as MockWindowRef;
+    blogDashboardPageService = TestBed.inject(BlogDashboardPageService);
     loaderService = TestBed.inject(LoaderService);
     blogDashboardBackendApiService = TestBed.inject(
       BlogDashboardBackendApiService);
     alertsService = TestBed.inject(AlertsService);
+    component.ngOnInit();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
   });
 
   it('should create', () => {
     expect(component).toBeDefined();
   });
 
-  it('should initialize', fakeAsync(() => {
+  it('should add subscriptions on initialization', () => {
+    spyOn(blogDashboardPageService.updateViewEventEmitter, 'subscribe');
+
+    component.ngOnInit();
+
+    expect(blogDashboardPageService.updateViewEventEmitter.subscribe)
+      .toHaveBeenCalled();
+  });
+
+  it('should initialize tab according to active tab', fakeAsync(() => {
+    spyOn(component, 'initMainTab');
+
+    component.ngOnInit();
+    expect(component.activeTab).toBe('main');
+    expect(component.initMainTab).toHaveBeenCalled();
+
+    blogDashboardPageService.navigateToEditorTabWithId('123456sample');
+    mockWindowRef.nativeWindow.onhashchange();
+    tick();
+
+    expect(component.activeTab).toBe('editor_tab');
+
+    blogDashboardPageService.navigateToMainTab();
+    mockWindowRef.nativeWindow.onhashchange();
+    tick();
+    expect(component.activeTab).toBe('main');
+    expect(component.initMainTab).toHaveBeenCalled();
+  }));
+
```

Comment:
I'm unable to understand what exactly is being tested here.
Can you place comments mentioning the setup, action, and test components?

Reply:
Broken into separate tests. PTAL.